examples/webApiDependencyInjection/Program.cs

```diff
@@ -0,0 +1,34 @@
+using k8s;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Load kubernetes configuration
+var kubernetesClientConfig = KubernetesClientConfiguration.BuildDefaultConfig();
+
+// Register Kubernetes client interface as sigleton
+builder.Services.AddSingleton<IKubernetes>(new Kubernetes(kubernetesClientConfig));
```

Comment:
The client is voluntarily registered as a sigleton to avoid reinitialising the HttpClient used within it for each request.
A few days ago I was discussing about using an httpclient factory to avoid the overhead of creating httpclient instances.
But for the moment there seems to be a problem with the watch method and its timeouts.

Imagine receiving 50 calls per second, how much time would be spent creating httpclients, how many resources (creation and gc)?

https://github.com/kubernetes-client/csharp/discussions/945#discussioncomment-3131066

Reply:
I am sorry, there might be a misunderstanding. I am **NOT** suggesting to change the lifetime of the service. I think register it as singleton is fine. Like you mentioned, since it is registered as a singleton service, once an instance is created by DI, injection will reuse existing instance and there won't be multiple creations. That's good.

What I proposed was a very small detail - dispose is not managed by DI in this code because the Kubernetes instance is created by dev, and it then it needs **the dev to dispose** it.

```csharp
// Provide an instance of Kubernetes as the argument.
// Since the instance is created by dev, it requires the dev to dispose it after use.
builder.Services.AddSingleton<IKubernetes>(new Kubernetes(kubernetesClientConfig));
```

A small improvement is to **let the DI to dispose it**:

```csharp
// Provide a delegate of Func<IServiceProvider, Kubernetes> as the argument.
// The service lifetime is still singleton:
builder.Services.AddSingleton<IKubernetes>(p => new Kubernetes(kubernetesClientConfig));
```

I am leaning toward to think that shall apply to any `IDisposable`.

Reply:
I prefer this way, this would let the di control the lifetime and also only create when necessary

``` c#
builder.Services.AddSingleton<IKubernetes>(_ => new Kubernetes(kubernetesClientConfig));
```

Reply:
I added discards variable placeholder to the examples.